app/services/irs_attempts_api/tracker.rb

```diff
@@ -1,20 +1,31 @@
 module IrsAttemptsApi
   class Tracker
-    attr_reader :session_id, :enabled_for_session
+    attr_reader :session_id, :enabled_for_session, :request, :user, :sp
 
-    def initialize(session_id:, enabled_for_session:)
-      @session_id = session_id
+    def initialize(session_id:, request:, user:, sp:, enabled_for_session:)
+      @session_id = session_id # IRS session ID
+      @request = request
+      @user = user
+      @sp = sp
       @enabled_for_session = enabled_for_session
     end
 
     def track_event(event_type, metadata = {})
       return unless enabled?
 
+      event_metadata = {
+        user_agent: request&.headers['User-Agent'],
```

Comment:
I am still trying to determine how careful we need to be about some of these attributes being potentially nil.